src/copy.jl

```diff
@@ -76,11 +81,26 @@ function copy_model(model::Model)
     caching_mode = backend(model).mode
     new_model = Model(caching_mode = caching_mode)
 
+    # At JuMP's level, filter_constraints should work with JuMP.ConstraintRef, 
+    # whereas MOI.copy_to's filter_constraints works with MOI.ConstraintIndex. 
+    if filter_constraints !== nothing
+        filter_constraints_user = filter_constraints
+        filter_constraints = (cref) -> begin
+            if cref isa MOI.ConstraintIndex
+                jump_cref = constraint_ref_with_index(model, cref)
+                return filter_constraints_user(jump_cref)
+            else
```

Comment:
Could you add a comment explaining why this `else` clause is needed ? How could `cref` be something else than a `ConstraintIndex` ?

Reply:
I get it, it's because you use it below, maybe do `moi_filter_constraints = (con_idx) -> ...` that only works with index, pass that to `copy_to` and for the object dictionary, use `filter_constraints` ?

Reply:
```Julia
    function moi_filter_constraints(cref::MOI.ConstraintIndex)
        jump_cref = constraint_ref_with_index(model, cref)
        return filter_constraints(jump_cref)
    end
    filter = filter_constraints !== nothing ? moi_filter_constraints : nothing
    
    index_map = MOI.copy_to(
        backend(new_model),
        backend(model),
        copy_names = true,
        filter_constraints = filter
    )
```

Reply:
@odow, I used your suggestion in 7601072.